components/repliq/repliq.app.mjs

```diff
@@ -1,11 +1,62 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "repliq",
-  propDefinitions: {},
+  propDefinitions: {
+    templateId: {
+      type: "string",
+      label: "Template ID",
+      description: "The ID of the template you want to deploy.",
+      async options() {
+        const templates = await this.listTemplates();
+        if (templates.error) return [];
+
+        return templates.map(({
+          id: value, name: label,
+        }) => ({
+          label,
+          value,
+        }));
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUrl() {
+      return "https://api.repliq.co/v2";
+    },
+    _headers() {
+      return {
+        Authorization: `Bearer ${this.$auth.api_key}`,
+      };
+    },
+    _makeRequest({
+      $ = this,
+      path,
+      ...opts
+    }) {
+      return axios($, {
+        url: this._baseUrl() + path,
+        headers: this._headers(),
+        ...opts,
+      });
+    },
+    getCreditsCount() {
+      return this._makeRequest({
+        path: "/getCreditsCount",
+      });
+    },
```

Comment:
So that the `$` variable can be passed in:
```suggestion
    getCreditsCount(opts = {}) {
      return this._makeRequest({
        path: "/getCreditsCount",
        ...opts
      });
    },
```